src/psbtv2.ts

```diff
@@ -0,0 +1,953 @@
+  PSBT_OUT_AMOUNT = "03",
+  PSBT_OUT_SCRIPT = "04",
+  PSBT_OUT_TAP_INTERNAL_KEY = "05",
+  PSBT_OUT_TAP_TREE = "06",
+  PSBT_OUT_TAP_BIP32_DERIVATION = "07",
+  PSBT_OUT_PROPRIETARY = "fc",
+}
+
+// Provided to friendly-format the PSBT_GLOBAL_TX_MODIFIABLE bitmask from
+// PsbtV2.PSBT_GLOBAL_TX_MODIFIABLE which returns PsbtGlobalTxModifiableBits[].
+enum PsbtGlobalTxModifiableBits {
+  INPUTS = "INPUTS", // 0b000000001
+  OUTPUTS = "OUTPUTS", // 0b000000010
+  SIGHASH_SINGLE = "SIGHASH_SINGLE", // 0b000000100
+}
+
+/* 
+Global Constants
+ */
+
+const PSBT_MAP_SEPARATOR = Buffer.from([0x00]);
+const BIP_32_NODE_REGEX = /(\/[0-9]+'?)/gi;
+const BIP_32_HARDENING_OFFSET = 0x80000000;
+const ERROR_PSBT_NOT_VALID = Error("Not a valid psbt");
+
+/* 
+Helper Functions
+*/
+
+// Return psbt as Buffer.
+function bufferize(psbt: string | Buffer): Buffer {
+  if (Buffer.isBuffer(psbt)) {
+    return psbt;
+  }
+
+  if (typeof psbt === "string") {
+    if (validateHex(psbt) === "") {
+      return Buffer.from(psbt, "hex");
+    }
+
+    if (validBase64(psbt)) {
+      return Buffer.from(psbt, "base64");
+    }
+  }
+
+  throw ERROR_PSBT_NOT_VALID;
+}
+
+// Some keytypes have keydata which allows for multiple unique keys of the same
+// keytype. Getters which return values from these keys should search and return
+// values from all keys of that keytype. This function matches on the first byte
+// of each key string (hex encoded) and returns all values associated with those
+// keys as an array of string (hex encoded) values.
+function getNonUniqueKeyTypeValues(
+  maps: Map<Key, Value> | Map<Key, Value>[],
+  keytype: KeyType
+) {
+  if (Array.isArray(maps)) {
+    // It's a set of input or output maps, so recursively check each map and set
+    // values.
+    const values: NonUniqueKeyTypeValue[][] = maps.map(
+      (map) =>
+        // TODO: Figure out a better way to type this
+        getNonUniqueKeyTypeValues(map, keytype) as NonUniqueKeyTypeValue[]
+    );
+
+    return values;
+  }
+
+  const map = maps; // Not an array
+  const values: NonUniqueKeyTypeValue[] = [];
+
+  for (const [key, value] of map.entries()) {
+    if (key.startsWith(keytype)) {
+      values.push({ key, value: value?.toString("hex") || null });
+    }
+  }
+
+  return values;
+}
+
+// A getter helper for optional keytypes which returns lists of values as hex
+// strings.
+function getOptionalMappedBytesAsHex(
+  maps: Map<Key, Value>[],
+  keytype: KeyType
+) {
+  return maps.map((map) => map.get(keytype)?.toString("hex") ?? null);
+}
+
+// A getter helper for optional keytypes which returns lists of values as
+// numbers.
+function getOptionalMappedBytesAsUInt(
+  maps: Map<Key, Value>[],
+  keytype: KeyType
+) {
+  return maps.map((map) => map.get(keytype)?.readUInt32LE() ?? null);
+}
+
+// Accepts a BIP0032 path as a string and returns a Buffer containing uint32
+// values for each path node.
+function parseDerivationPathNodesToBytes(path: string): Buffer {
```

Comment:
It's just not as good. I'd rather replace it since I already had built one which, imo, does a more complete job, but I was thinking that increases scope too much.

Reply:
right. got it!